hash merge causes crash with 0.12upgrade

<!--
Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

If your issue relates to a specific Terraform provider, please open it in the provider's own repository. The index of providers is at https://github.com/terraform-providers .
-->

### Terraform Version
<!---
Run `terraform -v` to show the version, and paste the result between the ``` marks below.

If you are not running the latest version of Terraform, please try upgrading because your issue may have already been fixed.
-->

```
Terraform v0.12.0-beta1
+ provider.aws v1.60.0-dev20190216h00-dev
+ provider.random v2.1.0
```

### Terraform Configuration Files
<!--
Paste the relevant parts of your Terraform configuration between the ``` marks below.

For large Terraform configs, please use a service like Dropbox and share a link to the ZIP file. For security, you can also encrypt the files using our GPG public key.
-->

```hcl
variable "aws_region" {
  description = "AWS region to launch servers."
  default     = "eu-central-1"
}

variable "tags" {
  description = "Tags for each component"
  type        = "map"

  default = {
    environment = "corp"
    team        = "IT"
    method      = "terraform"
    repo        = "Some repo"
  }
}
provider "aws" {
  region = "${var.aws_region}"
}
resource "aws_vpc" "corp" {
  cidr_block           = "${var.vpc_cidr}"
  enable_dns_hostnames = true

  tags = "${merge(var.tags, { Name = "corp-vpc" })}"
}
```

### Debug Output
<!--
Full debug output can be obtained by running Terraform with the environment variable `TF_LOG=trace`. Please create a GitHub Gist containing the debug output. Please do _not_ paste the debug output in the issue, since debug output is long.

Debug output may contain sensitive information. Please review it before posting publicly, and if you are concerned feel free to encrypt the files using the HashiCorp security public key.
-->
https://gist.github.com/mutt13y/cf5b05cbe516fa3f67ee7c6605ceb399

### Crash Output
panic: upgradeExpr on unsupported type <nil>

goroutine 1 [running]:
github.com/hashicorp/terraform/configs/configupgrade.upgradeExpr(0x0, 0x0, 0xc0002851b2, 0x6, 0x1, 0xc00028a690, 0xc000aa2e00, 0x1, 0x1, 0x2357120, ...)
	/opt/gopath/src/github.com/hashicorp/terraform/configs/configupgrade/upgrade_expr.go:587 +0x5646
github.com/hashicorp/terraform/configs/configupgrade.upgradeExpr(0x2596fa0, 0xc00032eeb0, 0xc0002851b2, 0x6, 0x1, 0xc00028a690, 0xc000135670, 0x4, 0x40, 0x0, ...)
	/opt/gopath/src/github.com/hashicorp/terraform/configs/configupgrade/upgrade_expr.go:60 +0x40e6
github.com/hashicorp/terraform/configs/configupgrade.upgradeExpr(0x24408e0, 0xc000ac3140, 0xc0002851b2, 0x6, 0x1, 0xc00028a690, 0x0, 0x0, 0x0, 0x0, ...)
	/opt/gopath/src/github.com/hashicorp/terraform/configs/configupgrade/upgrade_expr.go:38 +0x55bf
github.com/hashicorp/terraform/configs/configupgrade.normalAttributeRule.func1(0x24408e0, 0xc000ac3140, 0x0, 0x68, 0x5, 0x3, 0xc0003bb698, 0x4)
	/opt/gopath/src/github.com/hashicorp/terraform/configs/configupgrade/upgrade_body.go:33 +0x59
github.com/hashicorp/terraform/configs/configupgrade.attributeRule.func1(0xc000135340, 0xc0003bb7e0, 0xc, 0xc000ac30e0, 0xc000706a08, 0x1, 0xc0003bb698)
	/opt/gopath/src/github.com/hashicorp/terraform/configs/configupgrade/upgrade_body.go:249 +0x145
github.com/hashicorp/terraform/configs/configupgrade.upgradeBlockBody(0xc0002851b2, 0x6, 0xc0003bb7e0, 0xc, 0xc000135340, 0xc0004c3300, 0x3, 0x4, 0x0, 0x0, ...)
	/opt/gopath/src/github.com/hashicorp/terraform/configs/configupgrade/upgrade_native.go:462 +0xb51
github.com/hashicorp/terraform/configs/configupgrade.(*Upgrader).upgradeNativeSyntaxResource(0xc0006f7710, 0xc0002851b2, 0x6, 0xc000135340, 0x4d, 0xc0003bb611, 0x7, 0xc0003bb5f9, 0x4, 0xc000ac2ea0, ...)
	/opt/gopath/src/github.com/hashicorp/terraform/configs/configupgrade/upgrade_native.go:339 +0x832
github.com/hashicorp/terraform/configs/configupgrade.(*Upgrader).upgradeNativeSyntaxFile(0xc0006f7710, 0xc0002851b2, 0x6, 0xc0001dc000, 0x9e, 0x29e, 0xc00028a690, 0x0, 0x0, 0x280, ...)
	/opt/gopath/src/github.com/hashicorp/terraform/configs/configupgrade/upgrade_native.go:108 +0x890
github.com/hashicorp/terraform/configs/configupgrade.(*Upgrader).Upgrade(0xc0006f7710, 0xc0004b6ff0, 0xc000230190, 0x3, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/configs/configupgrade/upgrade.go:101 +0x49c
github.com/hashicorp/terraform/command.(*ZeroTwelveUpgradeCommand).Run(0xc0004d0300, 0xc00000a070, 0x0, 0x0, 0xc0002a23e0)
	/opt/gopath/src/github.com/hashicorp/terraform/command/012_config_upgrade.go:152 +0x829
github.com/mitchellh/cli.(*CLI).Run(0xc00028f040, 0xc00028f040, 0xc000427d20, 0x1)
	/opt/gopath/pkg/mod/github.com/mitchellh/cli@v1.0.0/cli.go:255 +0x207
main.wrappedMain(0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/main.go:223 +0xb37
main.realMain(0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/main.go:100 +0xb7
main.main()
	/opt/gopath/src/github.com/hashicorp/terraform/main.go:36 +0x2e



!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

### Expected Behavior
files are upgraded to 0.12 syntax
```
tags = merge(var.tags, { Name = "corp-vpc" })
```
### Actual Behavior
Crash

### Steps to Reproduce
Install AWS plugin in .terraform/plugins
terraform 0.12upgrade 


### Additional Context
<!--
Are there anything atypical about your situation that we should know? For example: is Terraform running in a wrapper script or in a CI system? Are you passing any unusual command line options or environment variables to opt-in to non-default behavior?
-->

if you remove the line
```
tags = "${merge(var.tags, { Name = "corp igw" })}"
```
there is no crash and the files are converted


### References
<!--
Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? For example:

- #6017

-->

--- a/configs/configupgrade/upgrade_expr.go
+++ b/configs/configupgrade/upgrade_expr.go
@@ -55,6 +55,7 @@ Value:
 					Detail:   fmt.Sprintf("Interpolation parsing failed: %s", err),
 					Subject:  hcl1PosRange(filename, tv.Pos).Ptr(),
 				})
+				return nil, diags
 			}
 
 			interpSrc, interpDiags := upgradeExpr(hilNode, filename, interp, an)

